core/server/sources.go

```diff
@@ -110,3 +110,54 @@ func (as *appServer) ListHelmRepositories(ctx context.Context, msg *pb.ListHelmR
 		HelmRepositories: results,
 	}, nil
 }
+
+func (as *appServer) AddHelmChart(ctx context.Context, msg *pb.AddHelmChartReq) (*pb.AddHelmChartRes, error) {
+	k8s, err := as.k8s.Client(ctx)
+	if err != nil {
+		return nil, doClientError(err)
+	}
+
+	src := types.ProtoToHelmChart(msg)
+
+	if err := k8s.Create(ctx, &src); err != nil {
+		return nil, status.Errorf(codes.Internal, "creating source for helm chart %q: %s", msg.Name, err.Error())
+	}
+
+	return &pb.AddHelmChartRes{
+		Success:   true,
+		HelmChart: types.HelmChartToProto(&src),
+	}, nil
+}
+
+func (as *appServer) ListHelmCharts(ctx context.Context, msg *pb.ListHelmChartReq) (*pb.ListHelmChartRes, error) {
+	k8s, err := as.k8s.Client(ctx)
+	if err != nil {
+		return nil, doClientError(err)
+	}
+
+	list := &sourcev1.HelmChartList{}
+
+	var opts client.MatchingLabels
```

Comment:
I think it might be a good time to turn this logic into a helper func, since we need to repeat it in several places. WDYT?